flare/env.py

```diff
@@ -32,40 +76,84 @@ def __init__(self, structure: Structure, atom: int, cutoffs, sweep=1):
         self.ctype = structure.coded_species[atom]
 
         self.cutoffs = np.copy(cutoffs)
+        self.cutoffs_mask = cutoffs_mask
+
+        self.setup_mask()
+
+        assert self.scalar_cutoff_3 <= self.scalar_cutoff_2, \
+            "2b cutoff has to be larger than 3b cutoff"
+        # # TO DO, once the mb function is updated to use the bond_array_2
+        # # this block should be activated.
+        # assert self.scalar_cutoff_mb <= self.scalar_cutoff_2, \
+        #         "mb cutoff has to be larger than mb cutoff"
 
         self.compute_env()
 
+    def setup_mask(self):
+
+        self.scalar_cutoff_2, self.scalar_cutoff_3, self.scalar_cutoff_mb, \
+            self.cutoff_2b, self.cutoff_3b, self.cutoff_mb, \
+            self.nspecie, self.n2b, self.n3b, self.nmb, self.specie_mask, \
+            self.bond_mask, self.cut3b_mask, self.mb_mask = \
```

Comment:
It's good practice to make sure all the attributes that the class could have are defined in the __init__ function, even if they are populated later. I see what you have in mind by wrapping it up in a tiny function but I think they can stay in the __init__.

Reply:
yes. I sort of implicitly do it. But it's more clear to set it up front in __init__

Reply:
See Zen of Python, PEP20, Line 2:
https://www.python.org/dev/peps/pep-0020/
`Explicit is better than implicit.` :)

Reply:
This is implemented in the new factory branch